opentracing/tracer.py

```diff
@@ -37,16 +39,73 @@ class Tracer(object):
 
     _supported_formats = [Format.TEXT_MAP, Format.BINARY, Format.HTTP_HEADERS]
 
-    def __init__(self):
+    def __init__(self, scope_manager=None):
+        self._scope_manager = ScopeManager() if scope_manager is None \
+                else scope_manager
         self._noop_span_context = SpanContext()
         self._noop_span = Span(tracer=self, context=self._noop_span_context)
+        self._noop_scope = Scope(self._scope_manager, self._noop_span)
+
+    @property
+    def scope_manager(self):
+        """ScopeManager accessor"""
+        return self._scope_manager
+
+    def start_active_span(self,
```

Comment:
Hey @indrekj 

So at this time we are using the same naming as the Java 0.31 API - I do remember, though, there was debate on this same item when testing out such API. IRC, it was something like:

```java
tracer.buildSpan("one").startActive();
tracer.buildSpan("two").startScope();
```

There was also some discussion about it in the Python side, and, all good and bad things considered, we decided to go with the `start_active_span()` name again. Hope that explains it ;)

Reply:
start_span_active might be less confusing. In Java the word span is not in the method name, so startActive and startScope were not significantly different, but here it does sound odd vs return type.

Reply:
I get the feeling `start_span_active` could still also lead to some confusion...

Opinion on this? @palazzem @pglombardo

Reply:
In ruby, it's `span = OpenTracing.start_active_span('operation_name')`.

I think there's a longer discussion about how to make scopes "hidden" from application developers, but still available to framework instrumentors via the ScopeManager. But I'd prefer we do that across-language, and not have a skewed API between languages at this stage.

Reply:
I'd highly suggest we stick to this API for now. We could later change it and deprecate it if feedback shows users are confused about it.

Reply:
I wouldn't take ruby as an example because we pretty much copied the python interface one-to-one. I started implementing scopes for ruby zipkin tracer and then noticed this method return value. That's why I was curios why it was chosen here.  I'm not aware of any ruby tracer implementations that are using scopes yet so might still be a good time to change it if needed. /cc ruby side @mwear @itsderek23

Reply:
Ack sorry, I meant scope.

If we're going to change it, we should change it in both places. If we can change it in Ruby without breaking anyone, then perhaps there's a window of opportunity.

If we change the return value to `span`, the scope can still be accessed via the scope manager. This feels cleaner to me. Essentially, the Tracer has `start_active_span` and `active_span` as convenience methods. Applications developers, in general, do not need to think about Scopes. For developers writing tricky instrumentation for frameworks, etc, they can use the ScopeManager directly.

Reply:
FYI it's not just Ruby, it's also Java, PHP, and C#. So there's a broader decision to make here. I'd prefer we think about how to address this in a cross language manner, and not have python (or python and ruby) be slightly different than the rest, even if it is slightly better. 

This probably means adding a new method to these other languages, and deprecating the current one. But let's check in with the Cross-Language Working Group before changing the API here, as it would essentially demand that the other languages update theirs.

Reply:
After some discussion in gitter, we'd like to go with `start_active_scope` for now. There's a broader discussion for how to give application developers a simpler interface for the common case, when they are creating active child spans without any fancy scope juggling. But that is for a future release, not this one!